blocks/algorithms/__init__.py

```diff
@@ -392,7 +393,8 @@ class Scale(StepRule):
 
     """
     def __init__(self, learning_rate=1.0):
-        self.learning_rate = shared_floatx(learning_rate)
+        self.learning_rate = shared_floatx(learning_rate, "learning_rate")
+        add_role(self.learning_rate, ALGORITHM_STATE)
```

Comment:
It seems to me that one role is not sufficient. One may want to distinguish between algorithm accumulators, like velocity in SGD with momentum, and algorithm hyperparameters, like the momentum it self. Arguably, these should be two different roles, but they might share a common ancestor. E.g. `ALGORITHM_HYPERPARAMETER` and `ALGORITHM_BUFFER` can be subroles of `ALGORITHM_STATE`.